src/components/Checkout/CheckoutAPICalls.tsx

```diff
@@ -1,44 +1,103 @@
 import { getRole } from '../contexts/UserContext';
-import { CheckoutItemProp, ClientPrincipal, ResidentInfo } from '../../types/interfaces';
+import {
+  CheckoutItemProp,
+  ClientPrincipal,
+  ResidentInfo,
+} from '../../types/interfaces';
 import { ENDPOINTS, API_HEADERS } from '../../types/constants';
 
-export async function processWelcomeBasket(user: ClientPrincipal | null, loggedInUserId: number, checkoutItems: CheckoutItemProp[], residentInfo: ResidentInfo) {
-  API_HEADERS['X-MS-API-ROLE'] = getRole(user);
-  const response = await fetch(ENDPOINTS.CHECKOUT_WELCOME_BASKET, {
-    method: 'POST',
-    headers: API_HEADERS,
-    body: JSON.stringify({
-      user_id: loggedInUserId,
-      mattress_size: checkoutItems[0].id,
-      quantity: checkoutItems[0].quantity,
-      resident_id: residentInfo.id,
-      message: "",
-    }),
-  });
-  return await response.json();
-}
+export async function processWelcomeBasket(
+  user: ClientPrincipal | null,
+  loggedInUserId: number,
+  checkoutItems: CheckoutItemProp[],
+  residentInfo: ResidentInfo,
+) {
+  try {
+    API_HEADERS['X-MS-API-ROLE'] = getRole(user);
+    const response = await fetch(ENDPOINTS.CHECKOUT_WELCOME_BASKET, {
+      method: 'POST',
+      headers: API_HEADERS,
+      body: JSON.stringify({
+        user_id: loggedInUserId,
+        mattress_size: checkoutItems[0].id,
+        quantity: checkoutItems[0].quantity,
+        resident_id: residentInfo.id,
+        message: '',
+      }),
+    });
 
-export async function processGeneralItems(user: ClientPrincipal | null, loggedInUserId: number, checkoutItems: CheckoutItemProp[], residentInfo: ResidentInfo) {
-  API_HEADERS['X-MS-API-ROLE'] = getRole(user);
-  const response = await fetch(ENDPOINTS.CHECKOUT_GENERAL_ITEMS, {
-    method: 'POST',
-    headers: API_HEADERS,
-    body: JSON.stringify({
-      user_id: loggedInUserId,
-      items: checkoutItems.map((item) => ({ id: item.id, quantity: item.quantity, additional_notes: item.additional_notes })),
-      resident_id: residentInfo.id,
-      message: "",
-    }),
-  });
-  return await response.json();
+    if (!response.ok) {
+      let errorMessage = 'Checkout failed';
+      try {
+        const errorData = await response.json();
+        errorMessage =
+          errorData.error?.message ||
+          `HTTP ${response.status}: ${response.statusText}`;
+      } catch {
+        errorMessage = `HTTP ${response.status}: ${response.statusText}`;
+      }
+      throw new Error(errorMessage);
+    }
+
+    return await response.json();
+  } catch (error) {
+    console.error('Error processing welcome basket checkout:', error);
```

Comment:
I think we should avoid console output in production code. Although this is error and not log, I don't think it adds anything to our diagnostics. (We propagate the error message and show it to the use in a snackbar). Unless you have a compelling argument, can we remove it?